x-pack/plugin/searchable-snapshots/src/main/java/org/elasticsearch/xpack/searchablesnapshots/cache/shared/FrozenCacheService.java

```diff
@@ -130,6 +141,55 @@ public void validate(final ByteSizeValue value, final Map<Setting<?>, Object> se
         Setting.Property.NodeScope
     );
 
+    public static final Setting<ByteSizeValue> SNAPSHOT_CACHE_SIZE_MAX_HEADROOM_SETTING = new Setting<>(
+        new Setting.SimpleKey(SHARED_CACHE_SETTINGS_PREFIX + "size.max_headroom"),
+        (settings) -> {
+            if (SNAPSHOT_CACHE_SIZE_SETTING.exists(settings) == false && isDedicatedFrozen(settings)) {
+                return "100GB";
+            }
+
+            return "-1";
+        },
+        (s) -> ByteSizeValue.parseBytesSizeValue(s, SHARED_CACHE_SETTINGS_PREFIX + "size.max_headroom"),
+        new Setting.Validator<>() {
+            private final Collection<Setting<?>> dependencies = List.of(SNAPSHOT_CACHE_SIZE_SETTING);
+
+            @Override
+            public Iterator<Setting<?>> settings() {
+                return dependencies.iterator();
+            }
+
+            @Override
+            public void validate(ByteSizeValue value) {
+                // ignore
+            }
+
+            @Override
+            public void validate(ByteSizeValue value, Map<Setting<?>, Object> settings, boolean isPresent) {
+                if (isPresent && value.getBytes() != -1) {
+                    RelativeByteSizeValue sizeValue = (RelativeByteSizeValue) settings.get(SNAPSHOT_CACHE_SIZE_SETTING);
+                    if (sizeValue.isAbsolute()) {
+                        throw new SettingsException(
+                            "setting [{}] cannot be specified for absolute [{}={}]",
+                            SNAPSHOT_CACHE_SIZE_MAX_HEADROOM_SETTING.getKey(),
+                            SNAPSHOT_CACHE_SIZE_SETTING.getKey(),
+                            sizeValue.getStringRep()
+                        );
+                    }
+                }
+            }
+        },
+        Setting.Property.NodeScope
+    );
+
+    private static boolean isDedicatedFrozen(Settings settings) {
```

Comment:
I'm surprised this method did not already exist.

Reply:
Maybe add this method to `DiscoveryNode`?

Reply:
I centralized this to `DiscoveryNode` in 616940c2d33525928f57575ebe61be90e921d73b